src/core/index.js

```diff
@@ -582,7 +582,9 @@ export function loginErrorMessage(m, error, type) {
     if (
       currentCaptcha && (
         currentCaptcha.get('provider') === 'recaptcha_v2' ||
-        currentCaptcha.get('provider') === 'recaptcha_enterprise'
+        currentCaptcha.get('provider') === 'recaptcha_enterprise' ||
+        currentCaptcha.get('provider') === 'hcaptcha' ||
+        captchaConfig.get('provider') === 'friendly_captcha'
```

Comment:
Is it expected to result in `invalid_recaptcha` when using `hcaptcha` or `friendly_captcha` ?

Reply:
Yes, this is used to display the appropriate localized error message which is consistent for all intents and purposes. Although I don't love the naming, this would require changes to all localization files in `i18n` which is a rather extensive change.